lib/google_search_data_viewer/keyword/google_search_parser.ex

```diff
@@ -0,0 +1,66 @@
+defmodule GoogleSearchDataViewer.Keyword.GoogleSearchParser do
+  @css_search_selectors %{
+    top_adwords: "#tads > .uEierd a.sVXRqc",
+    top_non_adwords: ".MhgNwc a",
+    non_adwords: ".yuRUbf a",
+    bottom_adwords: "#bottomads .uEierd a.sVXRqc"
+  }
+
+  def get_url_statistics(html) do
+    {_, parsed_html} = Floki.parse_document(html)
+
+    []
+    |> prepend_top_adwords(parsed_html)
+    |> prepend_top_non_adwords(parsed_html)
+    |> prepend_non_adwords(parsed_html)
+    |> prepend_bottom_adwords(parsed_html)
+  end
+
+  defp prepend_top_adwords(url_stats, parsed_html) do
+    urls =
+      parsed_html
+      |> Floki.find(@css_search_selectors.top_adwords)
+      |> Floki.attribute("href")
+
+    Enum.map(
+      urls,
+      fn url -> %{url: url, is_adword: true, is_top_adword: true} end
+    ) ++ url_stats
```

Comment:
How about using `Enum.concat/2`? So the whole function can be done in pipe operator
```suggestion
      parsed_html
      |> Floki.find(@css_search_selectors.top_adwords)
      |> Floki.attribute("href")
      |> Enum.map(fn url -> %{url: url, is_adword: true, is_top_adword: true} end) 
      |> Enum.concat(url_stats)
```

Reply:
Also, we can make use of the anonymous function in `Enum.map` to make the function shorter. Just my personal preference, feels free to ignore it. 😄
```elixir
Enum.map(&(%{url: &1, is_adword: true, is_top_adword: true})) 
```

Reply:
Completely forgot about `concat`, much better!

I see the use for the anonymous/captured function so thanks for reminding me about that but for this case it seems a bit more readable to not use it if that's OK.